src/librustdoc/html/render/mod.rs

```diff
@@ -2577,7 +2578,9 @@ fn render_call_locations(w: &mut Buffer, cx: &Context<'_>, item: &clean::Item) {
           <span></span>\
           <h5 id=\"{id}\" class=\"section-header\">\
              <a href=\"#{id}\">Examples found in repository</a>\
+             <a class=\"scrape-help\" href=\"{doc_prefix}/rustdoc/scraped-examples.html\" target=\"_blank\">?</a>\
```

Comment:
I think this is way too much:
 * It's pretty obvious what this is.
 * It'll increase the page size once more for a very minor gain.

Reply:
cc @jsha and @camelid who have thoughts on this

Reply:
I really don't think we should be worrying about a small increase in page size like this. And I don't think it's that obvious what it is. For you, it may be obvious, since you've been immersed in this feature throughout its development, but most users have never even heard of it.

Reply:
Even if they don't know about the feature, the text itself is pretty explicit: "Examples found in...". The rustdoc book exists and will include a chapter about this but I don't think there is enough interest for it.

Also, it's definitely not a small increase in page size. Especially on big crates with a lot of methods.

Reply:
> The rustdoc book exists and will include a chapter about this but I don't think there is enough interest for it.

I'm not sure what you mean about interest.

Reply:
Let me reword: I tried to say that since the rustdoc book exists and the feature itself is quite explicit, there is no point to include a link every time linking to the book chapter directly.

Now that I think about it, linking to the rustdoc book directly is also problematic for stable/beta/nightly links because the name of the chapter or its location could change whereas the link will stay around at a previous location

Reply:
> Now that I think about it, linking to the rustdoc book directly is also problematic for stable/beta/nightly links because the name of the chapter or its location could change whereas the link will stay around at a previous location

The code explicitly uses the correct channel, so that shouldn't be an issue in most cases. Since docs.rs uses nightly, it might not be solved there (since only the latest nightly is kept), but I don't think it's of concern.

Reply:
Let me show an example: you are using the current stable, 1.58, so the link is pointing to `doc.rust-lang.org/book/stable/this-chapter` (something along the line, the `/stable/` is the important part). Then a few months later, after a few updates in the books, the link changed but the URL still points to the previous one which is now invalid. Linking to external resources is generally a terrible idea.

If we really want to have some extra explanation, we need to find another way like having a popup with short explanation and saying to take a look at the rustdoc book, or something else which doesn't link directly to external resources.

Reply:
No, the link will be `/1.58.0/`, not `/stable/`.

Reply:
Not on docs.rs, which is literally the biggest website containing all crates documentation.

Reply:
Well, on docs.rs, it will be `/nightly/`, but I see your point. I still think it's unlikely for the docs to move much. If it becomes an issue, we could always change it.

Reply:
I think it's big enough of a potential problem to not be added. And there is also the size increase issue too.

Reply:
I still think clearly communicating about features like this is a good idea, but I don't want to argue about it; it's not important enough.

Reply:
I agree on the communication part. I don't agree on how it's currently done. Sorry for nitpicking on this but I think it's really important to have something working on all situations...

Reply:
@GuillaumeGomez if it's primarily page size you're worried about, I could always move the rendering logic to Javascript. I could append the `<a>` after the page loads.

Reply:
The page size is easy to fix indeed (I'd prefer to have it done on click only instead of generating it for all items, but that's another debate). My main worry is really about the link target.

Reply:
@willcrichton Like I said, instead of a link, maybe a small popup like the help menu which would say something like: "this was generated by the scrape-examples feature. You can find more information about it in the rustdoc book" where `rustdoc book` is a link to the rustdoc book (but not to the chapter to prevent this current issue).

The idea behind this is to name the feature so people know what to look for and provide them a link where they can find the information without worrying about the link being broken.

Does it sound good to you? (and to you too @camelid and @jsha?)

Reply:
@GuillaumeGomez I think you're proposing to have a tooltip with a clickable link inside it. That's a bad UI because it creates [hover tunnels](https://uxmovement.com/navigation/why-hover-menus-do-users-more-harm-than-good/).

Also, the problem you're trying to address is: the destination of the link might 404 in the future. You'd like to solve it by: making the link point to the whole book rather than a chapter. Whether we have a popup or a direct link is orthogonal to that question.

I'd like to solve that problem by: If the name of the chapter changes, put in a redirect. If mdbook doesn't support redirects, we should add it. They'd probably have to be JS redirects since mdbook just emits HTML, not server configs. But that's fine.

Reply:
I'm really not a big fan of that idea but that sounds like a good compromise.

Reply:
Ideally mdbook would also emit some sort of redirect map that can be consumed by server configs, so aware servers (like doc.rust-lang.org) can do a proper 302. But at a minimum it would need to support a JS redirect, in case a book is uploaded to an unconfigured server, or if it is being served locally.

Reply:
So is the resolution: we'll leave the code as-is, and deal with adding a redirect if the issue arises?

Reply:
I'd prefer to have mdbook update done before merging this because TODOs tend to remain as such. ^^'

Also, please generate this link using JS as you suggested. To get the current rustdoc version, you can use `getVar("rustdoc-version")` in the JS (you can see the result at the bottom of the help popup).

Reply:
@GuillaumeGomez I don't think it makes sense to hold up this feature on an mdbook feature, on the purely hypothetical idea that the chapter name will change. Note that this isn't the sort of TODO that will linger indefinitely. _If_ we rename the chapter, that will be the forcing function to implement redirects in mdbook.

How about this: In the Markdown source for mdbook, we add a comment: "Don't rename this chapter. If you do, make sure there's a redirect from the old name, so links from old documentation will continue to work."

Reply:
I have very cold feet on this. Linking to external resources on a stable feature is forever. If anything changes (from the domain name to the chapter name, pretty much anything), it's a dead link and we can't fix it. It's terrible user-wise.

Reply:
@GuillaumeGomez how about another possibility: What if we create a small, dedicated documentation page for the scrape examples and emit it with each documentation build? So we could link to scraped-examples.html within the current doc directory and be fully contained?

Reply:
I'm pretty sure mdBook already has a redirection feature. It's used by the rustc-dev-guide.

Reply:
@jsha: I think I prefer your solution a lot more. Like that, whether you have an internet connection or not, it'll work and we don't have to worry about the target being moved.

Reply:
If we're going to have an explainer in the docs that's distinct from the chapter in the rustdoc book, it should probably just be an overlay / popup like the current help menu. It wouldn't be more than a couple sentences probably.

Reply:
@jsha @GuillaumeGomez can y'all let me know what your final preference is for the scrape examples help message? I'd like to implement it this weekend.

Reply:
Writing a page and linking to it based on what @jsha suggested.

Reply:
Is it ok if I implement it as a popup though? I really think that would be both a better user experience and simpler for me to implement.

Reply:
Simpler I'm not sure. Adding an HTML page should be quite easy. Then you just have to link to it. Adding more JS for this doesn't seem the best solution but maybe I'm wrong?